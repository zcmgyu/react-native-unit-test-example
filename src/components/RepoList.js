import React from 'react';
import PropTypes from 'prop-types';
import { ListView, StyleSheet, Text } from 'react-native';

import RepoItem from './RepoItem';

const ds = new ListView.DataSource({
  rowHasChanged: (r1, r2) => r1.id !== r2.id
});

const RepoList = props => {
  const { repos, selected, selectRepo } = props;
  const dataSource = ds.cloneWithRows(repos);
  return (
    <ListView
      dataSource={dataSource}
      renderRow={repo => (
        <RepoItem
          isSelected={repo.id === selected}
          repo={repo}
          selectRepo={selectRepo}
        />
      )}
      contentContainerStyle={styles.list}
    />
  );
};

RepoList.propTypes = {
  repos: PropTypes.array.isRequired,
  selected: PropTypes.number,
  selectRepo: PropTypes.func
};

export const styles = StyleSheet.create({
  list: {
    paddingVertical: 40 // (48 - 8 of item padding)
  }
});

export default RepoList;
