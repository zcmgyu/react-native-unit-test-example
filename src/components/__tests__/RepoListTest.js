import React from 'react';
import RepoList from '../RepoList';
import mockedRepos from '../../../config/jest/dataMock';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders a RepoList using Snapshots', () => {
  const wrapper = renderer.create(
    <RepoList repos={mockedRepos} selectRepo={jest.fn()} />
  );
  expect(wrapper).toMatchSnapshot();
});
