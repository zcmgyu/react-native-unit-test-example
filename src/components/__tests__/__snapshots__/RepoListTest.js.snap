// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a RepoList using Snapshots 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingVertical": 40,
    }
  }
  dataSource={
    ListViewDataSource {
      "items": 3,
    }
  }
  renderRow={[Function]}
  renderScrollComponent={[Function]}
>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "paddingHorizontal": 24,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "paddingBottom": 4,
            }
          }
        >
          Test 1
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "paddingBottom": 8,
            }
          }
        >
          50000 stars
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "paddingHorizontal": 24,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "paddingBottom": 4,
            }
          }
        >
          Test 2
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "paddingBottom": 8,
            }
          }
        >
          10000 stars
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "paddingHorizontal": 24,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "paddingBottom": 4,
            }
          }
        >
          Test 3
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "paddingBottom": 8,
            }
          }
        >
          499 stars
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
